spykes/popvis.py

```diff
@@ -0,0 +1,435 @@
+                                        conditions=conditions, window=window, 
+                                        binsize=binsize, plot=False)
+
+            for cond_id in np.sort(psth['data'].keys()):
+
+                if cond_id not in all_psth['data']:
+                    all_psth['data'][cond_id] = list()
+ 
+                all_psth['data'][cond_id].append(psth['data'][cond_id]['mean'])
+                        
+
+        for cond_id in np.sort(all_psth['data'].keys()):
+            all_psth['data'][cond_id] = np.stack(all_psth['data'][cond_id])
+
+        if plot is True:
+
+            self.plot_heat_map(all_psth, conditions_names=conditions_names, 
+                colors=colors, width=width, height=height)
+
+        return all_psth
+
+
+    def plot_heat_map(self, psth_dict, cond_id=None, 
+                    conditions_names=None, sortby=None, sortorder='descend', 
+                    normalize=None, colors=['Blues', 'Reds', 'Greens'], 
+                    width=10, height=5):
+
+        """
+        Plots heat map for neuron population
+
+        Parameters
+        ----------
+        psth_dict : dictionary
+            
+            With keys: 'event', 'conditions', 'binsize', 'window', and 'data'.
+            
+            Each entry in psth['data'] is itself a dictionary with keys of
+            each cond_id that correspond to the means for that condition
+
+        cond_id
+            Which psth to plot indicated by the key in all_psth['data'].
+            If None then all are plotted.
+
+        conditions_names: str (or list)
+            Name(s) to appear in the title
+
+        sortby: str or list
+            None: 
+            'rate': sort by firing rate
+            'latency': sort by peak latency
+            list: list of integer indices to be used as sorting indicces
+
+        sortorder: direction to sort in
+            'descend'
+            'ascend'
+
+        normalize: str
+            None
+            'all' : divide all PSTHs by highest peak firing rate in all neurons
+            'each' : divide each PSTH by its own peak firing rate
+
+        colors: list of colors for heatmap
+
+        width: int
+            width of graph
+
+        height: int
+            height of graph
+
+        """
+
+        window = psth_dict['window']
+        binsize = psth_dict['binsize']
+        conditions = psth_dict['conditions']
+        event = psth_dict['event']
+
+
+        if conditions_names is None:
+            conditions_names = np.sort(psth_dict['data'].keys()).tolist()
+
+        if cond_id is None:
+            keys = np.sort(psth_dict['data'].keys())
+        else:
+            keys = cond_id
+
+        for i, cond_id in enumerate(keys):
+
+            # sort and norm the data
+
+            orig_data = psth_dict['data'][cond_id]
+
+            normed_data = self._get_normed_data(orig_data, normalize=normalize)
+
+            if isinstance(sortby, list):
+                sort_idx = sortby
```

Comment:
here, it might be good to check that `sortby` is basically a permutation of `range(0, len(neuron_list))`. one way to do this is:

```python
len(np.unique(sortby)) == len(sortby) and \
np.min(np.unique(sortby)) == np.min(sortby) and \
np.max(np.unique(sortby)) == np.max(sortby)
```

you can then raise an error if this criterion is not met with `raise ValueError()`